src/type-generator.ts

```diff
@@ -433,7 +437,13 @@ export class TypeGenerator {
   }
 
   private isPropertyRequired(property: string, structDef: JSONSchema4) {
-    return Array.isArray(structDef.required) && structDef.required.includes(property);
+    if (Array.isArray(structDef.required)) {
+      return structDef.required.find((name: string) => {
+        const normalized = camelCase(TypeGenerator.normalizeTypeName(name));
```

Comment:
Although the properties were being normalized and converted to camelCase, the required list was not matching that casing.

Reply:
See #478 - we would likely need to do this only if `camelCase` is enabled

Reply:
@eduardomourar I don't believe this is needed because the normalization only happens in the code we generate and this is looking up the property name based on the schema. Perhaps try to add a test that shows that this is indeed an issue?

Reply:
Okay, I figured what's the problem. We need to call this method with `originalName` instead of `name`. Fixing in another PR. Stay tuned.

Reply:
Alright. We could create a ES6 symbol to hold either the original name or the normalized one so that it can be used later.

Reply:
That's not required, we just need to pass the pre-normalized name to this method. I am fixing this as part of #480